### PR TITLE
Add Wentzel OK&VI transport cross section calculator

### PR DESCRIPTION
**Pull request by amandalund**
This adds a Wentzel OK&VI transport cross section calculator (which will be needed for the Wentzel VI MSC model). It also renames the `WentzelRatioCalculator` so it's now a helper class.

### CODE REVIEW

File: src/celeritas/em/xs/WentzelHelper.hh [hunk 1]
```diff
@@ -144,16 +144,17 @@ CELER_FUNCTION real_type WentzelRatioCalculator::calc_screening_coefficient(
...
-    return correction * screen_r_sq_elec() * sq_cbrt_z
+    return correction * this->screen_r_sq_elec() * sq_cbrt_z
            / value_as<MomentumSq>(particle.momentum_sq());
 }
 
 //---------------------------------------------------------------------------//
 /*!
- * Calculate the screening R^2 coefficient for incident electrons. This is
- * the constant prefactor of [PRM] eqn 8.51.
+ * Calculate the screening R^2 coefficient for incident electrons.
```
Comment (whokion) [inline src/celeritas/em/xs/WentzelHelper.hh]: It is unclear what `R^2` is referring to here.  Also better to put the `version` of RPM always as the eqn number may be changed the later if RPM is updated.

File: src/celeritas/em/xs/WentzelTransportXsCalculator.hh [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+ *
+ * \note This performs the same calculation as Geant4's
+ * G4WentzelOKandVIxSection::ComputeTransportCrossSectionPerAtom.
+ */
+class WentzelTransportXsCalculator
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using XsUnits = units::Native;  // [len^2]
```
Comment (whokion) [inline src/celeritas/em/xs/WentzelTransportXsCalculator.hh:32]: Will this be used elsewhere?

File: src/celeritas/em/xs/WentzelTransportXsCalculator.hh [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+ * OK and VI model in \c WentzelHelper and depend on properties of the incident
+ * particle, the energy cutoff in the current material, and the target element.
+ */
+CELER_FUNCTION
+WentzelTransportXsCalculator::WentzelTransportXsCalculator(
+    ParticleTrackView const& particle, WentzelHelper const& helper)
+    : z_(helper.atomic_number())
+    , screening_coeff_(helper.screening_coefficient())
+    , costheta_max_elec_(helper.costheta_max_electron())
+    , spin_betasq_(real_type(0.5) * particle.beta_sq())
```
Comment (whokion) [inline src/celeritas/em/xs/WentzelTransportXsCalculator.hh]: Is this related to the electron `spin` (0.5)? Okay, it seems that this is (`spin x beta^2`) even though the variable name is a somewhat artificial choice.

File: test/celeritas/em/Wentzel.test.cc [hunk 4]
```diff
@@ -212,6 +212,71 @@ TEST_F(CoulombScatteringTest, mott_xs)
...
+    ElementId elm_id{0};
+
+    AtomicNumber const z = this->material_params()->get(elm_id).atomic_number();
+
+    // Incident particle energy cutoff
+    MevEnergy const cutoff = this->cutoff_params()->get(mat_id).energy(
+        this->particle_track().particle_id());
+
+    std::vector<real_type> xs;
+    for (real_type energy : {50, 100, 200, 1000, 13000})
```
Comment (whokion) [inline test/celeritas/em/Wentzel.test.cc]: Is the test below 100MeV (Wenzel low limit) meaningful? Okay for testing as it should be a soft-limit. Is 13GeV special energy point? May add at least 100GeV and 1TeV?

**Review by whokion — APPROVED** [review]
This MR looks good to me! Just for my curiosity, why does xs (probably macroscopic xs based on this) need to be calculated on the fly for the Wenzel msc? Is this the special case when the Wenzel msc is replaced by the Coulomb single scattering within the model?

File: src/celeritas/em/xs/WentzelHelper.hh (hunk 1 above)
Comment (amandalund) [reply src/celeritas/em/xs/WentzelHelper.hh]: I tried to clarify the comment and added a PRM reference in the class description.

File: src/celeritas/em/xs/WentzelTransportXsCalculator.hh (hunk 2 above)
Comment (amandalund) [reply src/celeritas/em/xs/WentzelTransportXsCalculator.hh:32]: This is just to annotate that the calculated cross section is in native Celeritas units.

File: src/celeritas/em/xs/WentzelTransportXsCalculator.hh (hunk 3 above)
Comment (amandalund) [reply src/celeritas/em/xs/WentzelTransportXsCalculator.hh]: Right, it is spin x beta^2. Updated this to try to make it clearer.

File: test/celeritas/em/Wentzel.test.cc (hunk 4 above)
Comment (amandalund) [reply test/celeritas/em/Wentzel.test.cc]: I just used the same energies as the test for the Wentzel helper, so I'm not actually sure if there is a special meaning. I'll add some higher energy points.

**Comment by amandalund** [comment]
@whokion right, it's used to recalculate the macro xs in the calculation of the true step length to account for the single scattering.

**Comment by whokion** [comment]
@amandalund Thanks for all updates! @sethrj I think that this MR is good to be merged (unless @hhollenb has extra comments).

**Review by sethrj — APPROVED** [review]
Nice! Thanks.

**Pull request merged**